clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitResponse.java

```diff
@@ -192,7 +196,15 @@ public Builder merge(
             return this;
         }
 
-        public OffsetCommitResponse build() {
+        public OffsetCommitResponse build(short version) {
+            if (version >= 9) {
+                data.topics().forEach(topic -> {
+                    // Set the topic name to null if a topic ID for the topic is present.
+                    if (!Uuid.ZERO_UUID.equals(topic.topicId())) {
```

Comment:
Note: topic ID must not be null for a request/response version >= 9 to be serialized. `ZERO_UUID` means no topic ID specified.